Assets/Dependencies/SmashBrew/Stage/StageLedge.cs

```diff
@@ -1,26 +1,53 @@
 using HouraiTeahouse.SmashBrew.Characters;
 using UnityEngine;
+using UnityEngine.Networking;
 
 namespace HouraiTeahouse.SmashBrew.Stage {
 
     [AddComponentMenu("SmashBrew/Stage/Ledge")]
-    public class StageLedge : MonoBehaviour {
+    public class StageLedge : NetworkBehaviour {
 
         [SerializeField]
         bool _direction;
 
-        public bool Occupied { get; private set; }
+        public bool Direction
+        {
+            get { return _direction; }
+        }
+
+        [SyncVar, SerializeField]
+        bool _occupied;
+
+        public bool Occupied {
+            get { return _occupied; }
+            private set { _occupied = value; }
+        }
+
+        public NetworkIdentity GrabberId { get; private set; }
 
         void OnTriggerEnter(Collider collider) {
             if (!collider.CompareTag(Config.Tags.LedgeTag))
                 return;
             var movement = collider.GetComponentInParent<MovementState>();
             if (movement == null)
                 return;
-            movement.Direction = _direction;
-            movement.CurrentLedge = transform;
+            movement.tryLedgeGrab(this);
         }
 
+        public void Grab(MovementState movement) {
```

Comment:
This should have a null check. In C#, it's normal to throw a `System.ArgumentNullException` if a method is provided an invalid null argument.